doc/raster-overlays.md

```diff
@@ -0,0 +1,17 @@
+# RasterOverlayTileProvider
```

Comment:
I wrote this before I wrote any of the code in order to help clarify my own understanding of how it should work. I added it to the repo because I thought it was useful.